naga/src/lib.rs

```diff
@@ -597,6 +597,8 @@ bitflags::bitflags! {
         const LOAD = 0x1;
         /// Storage can be used as a target for store ops.
         const STORE = 0x2;
+        /// Storage can be used as a target for atomic ops.
+        const ATOMIC = 0x4;
```

Comment:
We have various bits of code in Naga that seem to assume that resources are not writable unless `STORE` is set. For example, see the handling of `Expression::GlobalVariable` in `FunctionInfo::process_expression`:

```rust
// storage data is only uniform when read-only
As::Storage { access } => !access.contains(crate::StorageAccess::STORE),
```

That doesn't seem like an unreasonable assumption, and I would expect other contributors to make it in the future. So it seems to me that we should probably require *both* `STORE` and `ATOMIC` to be set on an image in order to permit atomic operations on it.

I don't think we need to change the WGSL syntax; `texture_storage_2d<format, atomic>` can just build a Naga `TypeInner::Image { class: ImageClass::Storage { access: STORE | ATOMIC } }`.